platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/platformstatus/PlatformStatus.java

```diff
@@ -85,7 +86,11 @@ public enum PlatformStatus implements UniqueId {
      * <p>
      * NOTE: not in use
      */
-    CATASTROPHIC_FAILURE(11);
+    CATASTROPHIC_FAILURE(11),
+    /**
+     * The platform is done gossiping, and is in the process of writing a final freeze state to disk.
+     */
+    SAVING_FREEZE_STATE(12);
```

Comment:
I had put an updated diagram in the issue linked in the PR body, to avoid making modifications to an "accepted" proposal, and I see now that it warrants discussion as a team (Cody and I discussed it before)

The `SAVING_FREEZE_STATE` is necessary so that a final state can saved which includes the events which come to consensus while in the `FREEZING` state. At the same time, it doesn't make sense to have additional events after the last saved state. The process would be:
1. Gossip around signatures after entering `FREEZING`
2. After we've waiting however long for signatures to come in, stop gossip and finish writing a final state to disk
3. Shut down

It's true currently that the node doesn't stop gossiping during the freeze process: this would be modification to current behavior.

Reply:
I updated the diagram in the issue linked in the PR body with this change, since I didn't want to modify an "accepted" proposal. Cody and I had discussed this and I intended to bring it up with the team, but it slipped my mind.

@cody-littley correct me if I'm misrepresenting here, but I believe the logic we discussed is as follows:
After a freeze threshold has been crossed, the freeze state must be written to disk. However, it doesn't make any sense to write the state immediately, since signatures are being gathered on the freeze state. We will need to write the fully signed state to disk again anyway, once the signature gathering process is complete

Therefore, the process should look like this:
1. Once a freeze threshold has been crossed and the platform enters the `FREEZING` state, pause saving states to disk
2. Wait the configured amount of time in the `FREEZING` status, gathering signatures on the freeze state
3. Once the signature gathering time has elapsed, stop gossip (transition to `SAVING_FREEZE_STATE`)
4. Write a final signed state to disk
5. Shut down when done (transition to `FREEZE_COMPLETE`)

Reply:
I had intended to bring up this change with the team after discussing with Cody, but it slipped my mind. In another chat with him, we had considered even more necessary changes

Needless to say, I agree this warrants some more discussion with the size of the change

Reply:
Removing this intermediate state